azkaban-common/src/test/java/azkaban/project/JdbcProjectImplTest.java

```diff
@@ -218,11 +210,46 @@ public void testUpdatePermission() throws Exception {
         new Permission(Permission.Type.ADMIN), false);
 
-    final List<Triple<String, Boolean, Permission>> permissionsTriple = this.loader
-        .getProjectPermissions(project);
-    Assert.assertEquals(permissionsTriple.size(), 1);
-    Assert.assertEquals(permissionsTriple.get(0).getFirst(), "testUser1");
-    Assert.assertEquals(permissionsTriple.get(0).getThird().toString(), "ADMIN");
+    final Project sameProject = this.loader.fetchProjectByName("mytestProject");
+    Assert.assertEquals(sameProject.getUserPermissions().size(), 1);
+    Assert.assertEquals(sameProject.getUserPermissions().get(0).getFirst(), "testUser1");
+    Assert.assertEquals(sameProject.getUserPermissions().get(0).getSecond().toString(), "ADMIN");
+  }
+
+  @Test
+  public void testGetPermissionsOnAllActiveProjects() throws Exception {
+    createThreeProjects();
+    final Project project1 = this.loader.fetchProjectByName("mytestProject");
+    this.loader.updatePermission(project1, "testUser1", new Permission(Permission.Type.ADMIN), false);
+    this.loader.updatePermission(project1, "testUser2", new Permission(Permission.Type.EXECUTE), false);
+    this.loader.updatePermission(project1, "testGroup1", new Permission(Permission.Type.ADMIN), true);
+
+    final Project project2 = this.loader.fetchProjectByName("mytestProject2");
+    this.loader.updatePermission(project2, "testGroup2", new Permission(Permission.Type.READ), true);
+
+    final List<Project> projectList = this.loader.fetchAllActiveProjects();
+    final Project returnedProject1 = findProjectWithName(projectList, "mytestProject");
+    final Project returnedProject2 = findProjectWithName(projectList, "mytestProject2");
+    final Project returnedProject3 = findProjectWithName(projectList, "mytestProject3");
+
+    // Check to make sure project 1 and 2 were returned - but project 3 should not be returned (it has no associated
+    // permissions and we are using an INNER join).
+    Assert.assertNotNull(returnedProject1);
+    Assert.assertNotNull(returnedProject2);
+    Assert.assertNull(returnedProject3);
+
+    // Make sure proper permissions were returned for project1
+    Assert.assertTrue(returnedProject1.getUserPermission("testUser1").isPermissionSet(Permission.Type.ADMIN));
+    Assert.assertTrue(returnedProject1.getUserPermission("testUser2").isPermissionSet(Permission.Type.EXECUTE));
+    Assert.assertTrue(returnedProject1.getGroupPermission("testGroup1").isPermissionSet(Permission.Type.ADMIN));
+
+    // Make sure proper permissions were returned for project2
+    Assert.assertEquals(returnedProject2.getUserPermissions().size(), 0);
+    Assert.assertTrue(returnedProject2.getGroupPermission("testGroup2").isPermissionSet(Permission.Type.READ));
+  }
+
+  private Project findProjectWithName(List<Project> projects, String name) {
+    return projects.stream().filter(p -> p.getName().equals(name)).findFirst().orElse(null);
```

Comment:
I'll see your `findFirst()` and I raise you `findAny()`. Rumor has it it's marginally faster *when* there is a difference -- i.e. stream has encounter order *and* may have more than one element. Nit. Ignore me.

Reply:
😄